TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/Lane.cs

```diff
@@ -74,5 +75,25 @@ private float GetLaneLength()
             }
             return length;
         }
+
+        /// <summary>Get the length of the lane without intersections</summary>
+        public float GetLaneLengthNoIntersections()
+        {
+            float length = 0;
+            LaneNode curr = _start;
+            while(curr != null)
+            {
+                if(!curr.RoadNode.IsIntersection())
+                    length += curr.DistanceToPrevNode;
+                while(curr.RoadNode.IsIntersection())
```

Comment:
IsIntersection() only checks if the node is the actual intersection, junction edges will not return true from this function. Only the FourwayIntersection, threewayIntersection node will return true. Thus the while loop is useless and you could add an else after the if(!isIntersection) that just skips the node